api/src/backend/api/v1/views.py

```diff
@@ -1291,28 +1334,97 @@ def report(self, request, pk=None):
                     {"detail": "There is a problem with credentials."},
                     status=status.HTTP_403_FORBIDDEN,
                 )
-            file_content = s3_object["Body"].read()
-            filename = os.path.basename(output_location.split("/")[-1])
+            content = s3_obj["Body"].read()
+            filename = os.path.basename(key)
         else:
-            zip_files = glob.glob(output_location)
-            try:
-                file_path = zip_files[0]
-            except IndexError as e:
-                sentry_sdk.capture_exception(e)
```

Comment:
Why did you remove this?

Reply:
Because I made a refactor to avoid code repetition, now we check by using suffixes and patterns if the file exists or not:

```
keys = [obj["Key"] for obj in contents if obj["Key"].endswith(suffix)]
if not keys:
    return Response(
        {
            "detail": f"No compliance file found for name '{os.path.splitext(suffix)[0]}'."
        },
        status=status.HTTP_404_NOT_FOUND,
    )
```

Reply:
In that case, do we need to keep `sentry_sdk.capture_exception`?